lib/koans/15_protocols.ex

```diff
@@ -0,0 +1,39 @@
+defmodule Protocols do
+  use Koans
+
+  defprotocol School do
+    def enrol(name)
+  end
+
+  defimpl School, for: SecondarySchool do
+    def enrol(name) do
+      "#{name} enrolled at secondary school"
+    end
+  end
+
+  defimpl School, for: DanceSchool do
+    def enrol(name) do
+      "#{name} enrolled for dance"
+    end
+  end
+
+  defmodule EveningClass do end
+
+  @schools %{
+    :secondary => School.SecondarySchool,
+    :dance => School.DanceSchool,
+    :evening => EveningClass
+  }
+
+  koan "Commonality is the secret" do
+    school = Map.get(@schools, :dance)
```

Comment:
Using a map in this case actually feels like an unnecessary indirection. Any problem with just doing

```
School.DanceSchool.enroll("Emily")
```